### PR TITLE
aws publicip plugin test

### PR DESCRIPTION
**Pull request by wassoares**
- Implemented unit tests and fixes in the AwsV2PublicIpPlugin class;

### CODE REVIEW

**Comment by raonismaneoto** [comment]
PLEASE DON'T MERGE IT YET, @wassoares hasn't finished yet.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/aws/publicip/v2/AwsV2PublicIpPluginTest.java [hunk 1]
```diff
@@ -0,0 +1,678 @@
...
+	public void setUp() {
+		String awsConfFilePath = HomeDir.getPath() 
+				+ SystemConstants.CLOUDS_CONFIGURATION_DIRECTORY_NAME
+				+ File.separator 
+				+ CLOUD_NAME 
+				+ File.separator 
+				+ SystemConstants.CLOUD_SPECIFICITY_CONF_FILE_NAME;
+
+		this.plugin = Mockito.spy(new AwsV2PublicIpPlugin(awsConfFilePath));
+		this.sharedOrderHolders = Mockito.mock(SharedOrderHolders.class);
```
Comment (raonismaneoto) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/aws/publicip/v2/AwsV2PublicIpPluginTest.java]: why do u have to mock the sharedOrderHolders?

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/aws/publicip/v2/AwsV2PublicIpPluginTest.java [hunk 2]
```diff
@@ -0,0 +1,678 @@
...
+
+		// exercise
+		this.plugin.doCreateTagsRequests(key, value, resourceId, client);
+	}
+	
+	// test case: When calling the selectNetworkInterfaceFrom method, without anyone
+	// sub-net ID different from the default sub-net, it must return the network
+	// interface of default sub-net.
+	@Test
+	public void testSelectNetworkInterfaceUnsuccessful() throws FogbowException {
```
Comment (raonismaneoto) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/aws/publicip/v2/AwsV2PublicIpPluginTest.java]: shouldn't it return null? how the others plugins work?

Comment (wassoares) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/aws/publicip/v2/AwsV2PublicIpPluginTest.java]: In other plug-ins, such as OpenStack and CloudStack, public IP is attached using to the default network, which may be inaccurate compared to what we discussed earlier.

**Pull request merged**